SPA/public/js/ui.js

```diff
@@ -0,0 +1,173 @@
+};
+
+const refreshLinkedAccounts = (profile, supressEventSubscription = false) => {
+  const { user_id: primaryUserId, identities, email_verified, email } = profile;
+
+  const el = (selector) => document.querySelector(selector);
+
+  el("table.accounts tbody").remove();
+  const tbody = el("table.accounts").appendChild(
+    document.createElement("tbody")
+  );
+
+  const showInfoMessage = (msg) => {
+    const info = el("div.linking-message");
+    info.innerText = msg;
+  };
+
+  if (!email_verified) {
+    // don't offer linking for unverified emails.
+    showInfoMessage(
+      `The email ${email} is not verified. Account linking is only allowed for verified emails.`
+    );
+    el("button.btn-linkaccount").classList.add("hidden");
+    return;
+  }
+
+  const primary = (identity) =>
+    identity.provider !== primaryUserId.split("|")[0] ||
+    identity.user_id !== primaryUserId.split("|")[1];
+
+  const displayable = (identity) => ({
+    connection: identity.connection,
+    isSocial: identity.isSocial,
+    provider: identity.provider,
+    user_id: identity.user_id,
+    profileData: JSON.stringify(identity.profileData, null, 2),
+  });
+
+  const refresh = async () => {
+    showInfoMessage("");
+    const updatedProfile = await getUserProfile(primaryUserId);
+    refreshLinkedAccounts(updatedProfile, true);
+  };
+  if (identities.length > 1) {
+    identities
+      .filter(primary)
+      .map(displayable)
+      .forEach((identity) => {
+        const row = tbody.insertRow();
+        for (key in identity) {
+          const cell = row.insertCell();
+          cell.appendChild(document.createTextNode(identity[key]));
+        }
+        const actionCell = row.insertCell();
+        const unlinkButton = document.createElement("button");
+        unlinkButton.className = "btn btn-danger btn-unlinkaccount";
+        unlinkButton.innerText = "Unlink";
+        actionCell.appendChild(unlinkButton);
+        unlinkButton.addEventListener("click", async ({ target }) => {
+          try {
+            target.classList.add("disabled");
+            await unlinkAccount(identity);
+            refresh();
+          } finally {
+            target.classList.remove("disabled");
+          }
+        });
+      });
+  }
+  if (supressEventSubscription) return;
+  const linkButton = el("button.btn-linkaccount");
+  linkButton.addEventListener("click", async () => {
+    try {
+      await linkAccount();
+      refresh();
+    } catch ({ message }) {
+      showInfoMessage(message);
+    }
+  });
+};
+
+const updateUI = async () => {
+  try {
+    const isAuthenticated = await auth0.isAuthenticated();
+
+    if (isAuthenticated) {
+      const { sub: userId, ...user } = await auth0.getUser();
+      const profile = await getUserProfile(userId);
+      if (profile) refreshLinkedAccounts(profile);
+
+      document.getElementById("profile-data").innerText = JSON.stringify(
+        user,
```

Comment:
Can we use user.identites[0] instead of the user profile from the id_token? That way we can see to which connection does the primary identity belong to. 

If I have 3 linked identities (e.g. linkedin/google/facebook), Linkedin is primary, I login with Facebook, I see the profile (from Linkedin) and the two linked ones (google/facebook). I can't see I have a Linkedin one.

Reply:
![image](https://user-images.githubusercontent.com/1377205/80832930-fcf79d80-8c06-11ea-95e5-003ef6ecee53.png)

You mean the profile view?? ☝️

Reply:
https://github.com/auth0-samples/auth0-link-accounts-sample/pull/13/commits/098b674b0fc125fefe3902d3e0df4c6ed17a4ff4